org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractCodeMiningTest.java

```diff
@@ -92,45 +95,21 @@ protected String insertCodeMinings(XtextEditor editor) {
 			assertTrue("CodeMining is not resolved!", codeMining.isResolved());
 		}
 
-		codeMinings.sort((ICodeMining cm1, ICodeMining cm2) -> cm1.getPosition().getOffset() - cm2.getPosition().getOffset());
+		codeMinings.sort((ICodeMining cm1, ICodeMining cm2) -> cm2.getPosition().getOffset() - cm1.getPosition().getOffset());
 
-		int currentOffset = 0;
-		for (int i = 0; i < codeMinings.size(); i++) {
-
-			ICodeMining codeMining = codeMinings.get(i);
-
-			int codeMiningOffset = codeMining.getPosition().getOffset();
-
-			List<ICodeMining> codeMiningsOnTheSameOffset = getCodeMiningsByOffset(codeMinings, codeMiningOffset);
-
-			String codeMiningsText = getCodeMiningsText(codeMiningsOnTheSameOffset);
-
-			sb.append(text.substring(currentOffset, codeMiningOffset) + codeMiningsText);
-
-			currentOffset = codeMiningOffset;
-			if (containsLineHeaderCodeMining(codeMiningsOnTheSameOffset)) {
-				sb.append(System.lineSeparator());
-				currentOffset--;
+		Map<Integer, List<ICodeMining>> byPos = codeMinings.stream().collect(Collectors.groupingBy(e -> e.getPosition().getOffset(), LinkedHashMap::new, Collectors.toList()));
```

Comment:
Maybe more straight forward to use `Multimap.index(codeMinings, e->getPositions().getOffset())` but it's a matter of taste

Reply:
wanted to keep this similat to what eclipse does ;)